aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -1206,20 +1206,33 @@ public void onSuccess(CognitoUserSession userSession, CognitoDevice newDevice) {
                             @Override
                             public void getAuthenticationDetails(AuthenticationContinuation authenticationContinuation, String userId) {
                                 Log.d(TAG, "Sending password.");
+                                final HashMap<String, String> authParameters = new HashMap<>();
+                                // Check if the auth flow type setting is in the configuration.
+                                boolean authFlowTypeInConfig = false;
+                                if (awsConfiguration.optJsonObject(AUTH_KEY) != null &&
+                                    awsConfiguration.optJsonObject(AUTH_KEY).has("authenticationFlowType")) {
+                                    authFlowTypeInConfig = true;
+                                }
+
                                 try {
-                                    if (
-                                            awsConfiguration.optJsonObject(AUTH_KEY) != null &&
-                                            awsConfiguration.optJsonObject(AUTH_KEY).has("authenticationFlowType") &&
-                                            awsConfiguration.optJsonObject(AUTH_KEY).getString("authenticationFlowType").equals("CUSTOM_AUTH")
-                                    ) {
-                                        final HashMap<String, String> authParameters = new HashMap<String, String>();
+                                    // If there's a value in the config and it's CUSTOM_AUTH
+                                    if (authFlowTypeInConfig &&
+                                        awsConfiguration.optJsonObject(AUTH_KEY).getString("authenticationFlowType").equals("CUSTOM_AUTH")) {
```

Comment:
`"CUSTOM_AUTH".equals(otherstuff)` to protect against `null`. What happens if value `null` is saved as value? would `getString()` just throw since it's not a string?